bootstrap/src/main.ts

```diff
@@ -12,38 +12,56 @@
 import { BootstrapManager } from './bootstrap/bootstrap-manager'
 export { BootstrapManager } from './bootstrap/bootstrap-manager'
 
-// FIXME: Code duplication with single-app window manager.
-//   Should be fixed by introducing of routing (MVD-1535)
-function parseQuery() {
-  const queryString: string = location.search.substr(1);
-  const queryObject: { [id: string]: string } = {};
-  queryString.split('&').forEach(function(part) {
-    const pair = part.split('=').map(x => decodeURIComponent(x));
-    queryObject[pair[0]] = pair[1];
-  });
-  return queryObject;
-}
-const query = parseQuery();
-if (typeof query.pluginId === 'string') {
-  console.log(`ZWED5003I - Simple container requested with pluginId ${query.pluginId}`);
-  (window as any)['GIZA_SIMPLE_CONTAINER_REQUESTED'] = true;
-  (window as any)['GIZA_PLUGIN_TO_BE_LOADED'] = query.pluginId;
-  (window as any)['ZOWE_SWM_SHOW_LOGIN'] = query.showLogin;
-  (window as any)['GIZA_ENVIRONMENT'] = 'MVD';
-}
+processApp2AppArgs();
 
 try {
-  const simpleContainerRequested = (window as any)['GIZA_SIMPLE_CONTAINER_REQUESTED'];
-  const uriBroker = (window as any)['GIZA_ENVIRONMENT'];
+  const simpleContainerRequested = window['GIZA_SIMPLE_CONTAINER_REQUESTED'];
+  const uriBroker = window['GIZA_ENVIRONMENT'];
 
-  if (!simpleContainerRequested || uriBroker.toUpperCase() === 'MVD') {
+  if (!simpleContainerRequested || (uriBroker && uriBroker.toUpperCase() === 'MVD')) {
     BootstrapManager.bootstrapDesktopAndInject();
   }
 } catch (error) {
   console.error("ZWED5007E - Unable to bootstrap desktop!!");
   console.error(error);
 }
 
+/* Minor code duplication of StartURLManager, but Typescript gives compile problems we can't easily ignore when we import it */
+function processApp2AppArgs(url?: string): void {
+  const queryString = url || location.search.substr(1);
+  let pluginId, windowManager: any;
+
+  queryString.split('&').forEach(part => {
+    const [key, value] = part.split('=').map(v => decodeURIComponent(v));
+    switch (key) {
+      case "pluginId":
+        window['GIZA_PLUGIN_TO_BE_LOADED'] = value;
+        window['GIZA_SIMPLE_CONTAINER_REQUESTED'] = true;
+        window['GIZA_ENVIRONMENT'] = 'MVD';
+        pluginId = value;
+        break;
+      case "showLogin":
+        if (typeof value == "string") {
```

Comment:
The value is always a `string` because you parse a URL which is a `string`. Perhaps `window.ZOWE_SWM_SHOW_LOGIN` should be a `boolean`, and `showLogin` URL parameter should be a string with `true` and `false` as possible values;

Reply:
It really doesn't matter much to me, both work either way. Had a conditional for TS purposes

I can change it to boolean to make the URL more human readable